python/dgl/mock_sparse2/sparse_matrix.py

```diff
@@ -21,6 +21,33 @@ def val(self) -> torch.Tensor:
         """
         return self.c_sparse_matrix.val()
 
+    @val.setter
+    def val(self, x: torch.Tensor):
+        """Set the non-zero values inplace.
+
+        Parameters
+        ----------
+        x : torch.Tensor, optional
+            The values of shape (nnz) or (nnz, D)
+        """
+        self.c_sparse_matrix.set_val(x)
+
+    def __call__(self, x: torch.Tensor):
```

Comment:
I think it's my bad to include this in the first MVP. The more I think of it the more I feel the syntax is not straightforward, e.g., it may confuse users with functions. How about we remove it and come back to this after we see this is frequently used?

Reply:
Sure.